internal/scorecard/plugins/labels_test.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2019 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scplugins
+
+import (
+	"k8s.io/apimachinery/pkg/labels"
+	"testing"
+)
+
+func TestBasicShortNames(t *testing.T) {
+	cases := []struct {
+		name          string
+		testsSelected int
+		wantError     bool
+	}{
+		{"test=" + specBlockExists, 1, false},
+		{"test in (" + specBlockExists + ")", 1, false},
+		{"test in (" + specBlockExists + "," + writeIntoCR + ")", 2, false},
+		{"test=" + statusBlockExists, 1, false},
+		{"test=" + writeIntoCR, 1, false},
+		{"testXwriteintocr", 0, false},
+		{"test X writeintocr", 0, true},
+	}
+
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			var selector labels.Selector
+			selector, err := labels.Parse(c.name)
+			if err != nil && c.wantError {
+				t.Logf("Wanted error and got error : %v", err)
+				return
+			} else if err != nil && !c.wantError {
+				t.Errorf("Wanted result but got error: %v", err)
+				return
+			}
+
+			basicTests := NewBasicTestSuite(BasicTestConfig{})
+			basicTests.ApplySelector(selector)
+			testsSelected := len(basicTests.Tests)
+			if testsSelected != c.testsSelected {
+				t.Errorf("Wanted testsSelected %d, got: %d", c.testsSelected, testsSelected)
+			}
+		})
+
+	}
+}
+
+func TestOLMShortNames(t *testing.T) {
+	cases := []struct {
+		name          string
+		testsSelected int
+		wantError     bool
+	}{
+		{"test=" + bundleValidation, 1, false},
+		{"test=" + crdValidationSection, 1, false},
+		{"test=" + crdHasResources, 1, false},
+		{"test=" + specDescriptors, 1, false},
+		{"test=" + statusDescriptors, 1, false},
+		{"testXstatusdescriptors", 0, false},
+		{"test X statusdescriptors", 0, true},
+	}
+
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			var selector labels.Selector
+			selector, err := labels.Parse(c.name)
+			if err != nil && c.wantError {
+				t.Logf("Wanted error and got error : %v", err)
+				return
+			} else if err != nil && !c.wantError {
+				t.Errorf("Wanted result but got error: %v", err)
+				return
+			}
```

Comment:
Non-blocking nit: going forward, we should be using [assert](https://godoc.org/github.com/stretchr/testify/assert) to check errors and compare return values.

Reply:
IHMO we should NOT use assert and we should do the test as it was done here by using the standard test-lib provided by go. 
See: https://www.philosophicalhacker.com/post/integration-tests-in-go/

NOTE; I the past I was in favor of assert as well. However, note that it goes against the Golang standards and because of this is not added by default. Also, IDE's can be very helpful and give all structure implementation using the Golang lib and we do not need to add + 1 module as dep of the project just to do tests as well.

Reply:
We use `assert` in other parts of the SDK because it is succinct and implements comparison logic for us, and prints stack traces and diffs. It augments the standard library testing framework, not replaces it.

I'm not a fan of the IDE argument because everyone uses different IDE configurations, or no IDE at all. Instead, we should strive to codify low-LOC testing solutions that fit our needs, not necessarily follow what bloggers have to say about testing code.

A lot of really good code is not part of the standard library, and yet people import that code all the time. Pulling in one test dependency is trivial (albeit dependency blowup _can_ be an issue).

Reply:
Hi @estroz, 

Thank you for your input. Following some comments regards your reply just to share.

The blog shared has a good public summary of the info, however, you will find the same in the book [The Go Programa Lanaguage](https://www.gopl.io/ch1.pdf) and section `11.2.5. Writing Effective Tests` for example.  Also, you can see [here](https://www.gopl.io/) who are its authors. 

Regards the IDE, it is not mandatory at all but it is just a facility for who has it. And then, about the assert features, btw really tks for share, may the project has something very specific that you were able to already check, however, as far as I know, do diffs and print stack traces also can be done with the testing one.

Anyway, my vote still -1 to use assert as always as possible sorry. However, I am ok with this decision as well. It is not the biggest deal at all too  :-)

Reply:
Hi @estroz,

Thank you for your input. Following some comments regards your reply just to share.
The blog shared has a good public summary of the info, however, you will find the same in the book `The Go Programa Lanaguage` and section `11.2.5. Writing Effective Tests` for example.  Also, you can see [here](https://www.gopl.io/) who are its authors. Regards the IDE, it is not mandatory at all but it is just a facility for who has it. 

Anyway, my vote still -1 to use assert as always as possible sorry. However, I am ok with this decision as well. It is not the biggest deal at all too  :-)

Reply:
My two cents is that we _should_ use a test library, because it simplifies test code and makes it easier to read and write tests. In the past I've used `testify/assert` and I've found it to be productive and easy to use.

Another option is to pair `ginkgo` and `gomega`, which is what kubebuilder and controller-runtime are using.

I don't have a strong opinion on what we go with, but we should make a choice for the project and stick with it.

Reply:
Hi @joelanford,

IHMO: At the first moment, it simplifies because we are familiarised with assert from other languages. However, speaking about my own personal experience, after you effectively start to use the testing you do not want to move forward to use another one. :-) 

Also, note that in the book they explain why the Golang has not the assert as well and the idea behind the testing framework provide by the language. 

I'd also not vote in prol of `ginkgo` and `gomega` since are 3rd party libs not officially support. (can you imagine the pain to re-write all tests because the lib will be deprecated?) Also, IHMO we should adopted and provide to the users in the test-framework what is the Golang recommendation and standard instead of 3rd party one and we should try to reduce the dependencies with 3d-party instead of increase it in the project.

Reply:
Punting on this for now